samples/net/sockets/echo_server/src/tunnel.c

```diff
@@ -159,3 +159,13 @@ int init_tunnel(void)
 
 	return 0;
 }
+
+static int ignore_tunnel(void)
+{
+	net_conn_mgr_ignore_l2(&NET_L2_GET_NAME(VIRTUAL));
+
+	return 0;
+}
+
+BUILD_ASSERT(CONFIG_NET_CONNECTION_MANAGER_PRIORITY > 0);
+SYS_INIT(ignore_tunnel, APPLICATION, CONFIG_NET_CONNECTION_MANAGER_PRIORITY - 1);
```

Comment:
I find this part a bit unfortunate. This is because you need to explicitly know to add this for a given interface type. How do you, as a developer, will know that it's needed?

Perhaps a method with an interface or L2 flag, which indicates if an interface is suitable for watching by `conn_mgr`, would make sense. This could be added in a subsequent PR. This way we shift the responsibility from the developer to integration author.

Reply:
I'm not sure the responsibility should be with the integration provider; I can imagine a scenario where, for instance, where a connectivity-capable iface might be occasionally available, but the application developer does not want to rely on it. I also think that whether an iface is suitable for monitoring depends largely on context, and that context is defined in the application domain.

But I think I agree with you partially, that `net_conn_mgr_ignore_l2` is not really a great way for applications to control which ifaces they receive. The L2 type is often very disconnected from the actual nature of the iface. What we really need (and have discussed in other PRs) is some kind of (possibly extensible) tagging system for ifaces that is independent of the L2, and somehow communicates the underlying nature of each iface. Then we could allow applications to ignore/unignore by tag. This would also fix the responsibility issue by making it the responsibility of the integration author to accurately tag the ifaces they provide, while still allowing the application developer full control over what tags they are interested in.

But the conclusion we have come to in those discussions is that such a tagging system is beyond the scope of this initial implementation.

I mainly included `net_conn_mgr_ignore_l2` specifically because of this sample -- it relies on filtering by L2 type, and that is not possible with the new event behaviors. If we had preserved the old event behavior and added new events with the new behavior, this would not be necessary -- but that was the approach taken in https://github.com/zephyrproject-rtos/zephyr/pull/53118, and that approach did not stand up to review

Reply:
I start to wonder, is `SYS_INIT` configuration even needed here? IPIP interfaces have `NET_IF_NO_AUTO_START` set, so they won't start at boot, and we have `init_tunnel()` here which configures IP addresses, so this interface should not be considered "ready" before that. Especially, that the carrier state of an virtual interface is controlled by the operational UP status of the underlying interface - so the virtual iface should not be set to oper UP before the real one.

So I think the logic could be simplified a bit, and the interface could be ignored simply in `init_tunnel()`.

Reply:
Ah! Good point! I will switch to the approach you suggest